aws_lambda_powertools/tracing/tracer.py

```diff
@@ -703,7 +742,7 @@ def _add_response_as_metadata(
         self,
         method_name: Optional[str] = None,
         data: Optional[Any] = None,
-        subsegment: Optional[BaseSegment] = None,
+        subsegment: Optional[BaseSpan] = None,
```

Comment:
BaseSpan is not compatible with BaseSegment, so I would say this is a breaking change. BUT since it's a "private" method, I think it's ok. Just wanted you to double check it's fine.

Reply:
Yes this is tested.